fnf/data/geocode.py

```diff
@@ -0,0 +1,118 @@
+import logging
+import requests
+import numpy as np
+
+FIND_PLACE = "https://maps.googleapis.com/maps/api/place/findplacefromtext/json?"
+PLACE_DETAILS = "https://maps.googleapis.com/maps/api/place/details/json?"
+
+
+def place_by_name(place, key, FIND_PLACE=FIND_PLACE):
+    """Finds a Google Place ID by searching with its name.
+
+    Args:
+        place (str): Name of the place. It can be a restaurant, bar, monument,
+            whatever you would normally search in Google Maps.
+        key (str): Key for the Google API.
+        FIND_PLACE (str): Endpoint for the Google Places API. Note that the
+            service must be enabled in order to use it.
+
+    Returns:
+        (str) Place ID.
+    
+    """
+    params = {
+        "input": "{}".format(place),
+        "fields": "place_id",
+        "inputtype": "textquery",
+        "key": key,
+    }
+
+    r = requests.get(FIND_PLACE, params=params)
+    r.raise_for_status()
+
+    return r.json()["candidates"][0]["place_id"]
```

Comment:
What is the behaviour when the API fails to find any candidates? Will `[0]` result in an IndexError?

Reply:
Yes, which will be caught in the main script: https://github.com/nestauk/fnf/blob/4-geocode-places/fnf/geocode_affiliations.py#L52-L56

Shall I move the `try: except` block in `place_by_name` and return `None` when the API fails to find a match?

Reply:
These are the changes I'm thinking:
In `geocode.py`:
```python
    try:
        return r.json()["candidates"][0]["place_id"]
    except IndexError as e:
        logging.info(f"Failed to find a match for {place}")
        return None
```

and in `geocode_affiliations.py`:

```python
        r = place_by_name(name, key)
        if r:
            response = place_by_id(r, key)
            place_details = parse_response(response)
            place_details.update({"affiliation_id": id})
            s.add(AffiliationLocation(**place_details))
            s.commit()
        else:
            continue
```

Reply:
Yeah, this is nicer.

Note: In this sort of scenario one thing that can enhance readability is `if r is not None`, this  helps someone reading the code get a better idea of what condition you are checking (if `r` is not None, rather than if `r` is `True`).